pkg/sql/opt/exec/execbuilder/tests/local/local_test: TestExecBuild_collated_strings failed

pkg/sql/opt/exec/execbuilder/tests/local/local_test.TestExecBuild_collated_strings [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/13603860?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/13603860?buildTab=artifacts#/) on master @ [c74eac56a10e82798d8486d7aca3204a117bc8cf](https://github.com/cockroachdb/cockroach/commits/c74eac56a10e82798d8486d7aca3204a117bc8cf):


```
             ├── key: (1)
             ├── fd: ()-->(2)
             ├── distribution: test
             ├── prune: (1)
             ├── scan t98400
             │    ├── columns: k:1 s:2
             │    ├── stats: [rows=13, distinct(1)=13, null(1)=0, distinct(2)=3, null(2)=0]
             │    │   histogram(1)=  0  1  0  1  0  1  0  1  0  1  0  1  0  1  0  1  0  1  0  1   0  1   0  1   0  1
             │    │                <--- 1 --- 2 --- 3 --- 4 --- 5 --- 6 --- 7 --- 8 --- 9 --- 10 --- 11 --- 12 --- 13
             │    │   histogram(2)=  0                 1                 10                  2
             │    │                <--- 'foo' COLLATE en_US_u_ks_level2 ---- 'world' COLLATE en_US_u_ks_level2
             │    ├── cost: 42.86
             │    ├── key: (1)
             │    ├── fd: (1)-->(2)
             │    ├── distribution: test
             │    └── prune: (1,2)
             └── filters
                  └── s:2 = 'hello' COLLATE en_US_u_ks_level2 [outer=(2), constraints=(/2: [/'hello' COLLATE en_US_u_ks_level2 - /'hello' COLLATE en_US_u_ks_level2]; tight), fd=()-->(2)]
        but found (query options: "") :
            select
             ├── columns: k:1 s:2
             ├── stats: [rows=10, distinct(2)=1, null(2)=0]
             │   histogram(2)=  0                 10
             │                <--- 'hello' COLLATE en_US_u_ks_level2
             ├── cost: 64.41
             ├── key: (1)
             ├── fd: ()-->(2)
             ├── distribution: test
             ├── prune: (1)
             ├── scan t98400
             │    ├── columns: k:1 s:2
             │    ├── stats: [rows=13, distinct(1)=13, null(1)=0, distinct(2)=3, null(2)=0]
             │    │   histogram(1)=  0  1  0  1  0  1  0  1  0  1  0  1  0  1  0  1  0  1  0  1   0  1   0  1   0  1
             │    │                <--- 1 --- 2 --- 3 --- 4 --- 5 --- 6 --- 7 --- 8 --- 9 --- 10 --- 11 --- 12 --- 13
             │    │   histogram(2)=  0                 1                 10                  2
             │    │                <--- 'foo' COLLATE en_US_u_ks_level2 ---- 'world' COLLATE en_US_u_ks_level2
             │    ├── cost: 64.25
             │    ├── key: (1)
             │    ├── fd: (1)-->(2)
             │    ├── distribution: test
             │    └── prune: (1,2)
             └── filters
                  └── s:2 = 'hello' COLLATE en_US_u_ks_level2 [outer=(2), constraints=(/2: [/'hello' COLLATE en_US_u_ks_level2 - /'hello' COLLATE en_US_u_ks_level2]; tight), fd=()-->(2)]
            
[22:52:06] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/13652/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/collated_strings with config local: 3 tests, 1 failures
[22:52:07] --- total progress: 3 statements
--- total: 3 tests, 1 failures
    logic.go:4191: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/7c0b99e5fac8a43429506c705d588ce2/logTestExecBuild_collated_strings353617780
--- FAIL: TestExecBuild_collated_strings (2.15s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExecBuild_collated_strings.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-35428

pkg/sql/opt/exec/execbuilder/tests/local/local_test.TestExecBuild_collated_strings [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/13604920?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/13604920?buildTab=artifacts#/) on master @ [c74eac56a10e82798d8486d7aca3204a117bc8cf](https://github.com/cockroachdb/cockroach/commits/c74eac56a10e82798d8486d7aca3204a117bc8cf):


```
             ├── key: (1)
             ├── fd: ()-->(2)
             ├── distribution: test
             ├── prune: (1)
             ├── scan t98400
             │    ├── columns: k:1 s:2
             │    ├── stats: [rows=13, distinct(1)=13, null(1)=0, distinct(2)=3, null(2)=0]
             │    │   histogram(1)=  0  1  0  1  0  1  0  1  0  1  0  1  0  1  0  1  0  1  0  1   0  1   0  1   0  1
             │    │                <--- 1 --- 2 --- 3 --- 4 --- 5 --- 6 --- 7 --- 8 --- 9 --- 10 --- 11 --- 12 --- 13
             │    │   histogram(2)=  0                 1                 10                  2
             │    │                <--- 'foo' COLLATE en_US_u_ks_level2 ---- 'world' COLLATE en_US_u_ks_level2
             │    ├── cost: 42.86
             │    ├── key: (1)
             │    ├── fd: (1)-->(2)
             │    ├── distribution: test
             │    └── prune: (1,2)
             └── filters
                  └── s:2 = 'hello' COLLATE en_US_u_ks_level2 [outer=(2), constraints=(/2: [/'hello' COLLATE en_US_u_ks_level2 - /'hello' COLLATE en_US_u_ks_level2]; tight), fd=()-->(2)]
        but found (query options: "") :
            select
             ├── columns: k:1 s:2
             ├── stats: [rows=10, distinct(2)=1, null(2)=0]
             │   histogram(2)=  0                 10
             │                <--- 'hello' COLLATE en_US_u_ks_level2
             ├── cost: 64.41
             ├── key: (1)
             ├── fd: ()-->(2)
             ├── distribution: test
             ├── prune: (1)
             ├── scan t98400
             │    ├── columns: k:1 s:2
             │    ├── stats: [rows=13, distinct(1)=13, null(1)=0, distinct(2)=3, null(2)=0]
             │    │   histogram(1)=  0  1  0  1  0  1  0  1  0  1  0  1  0  1  0  1  0  1  0  1   0  1   0  1   0  1
             │    │                <--- 1 --- 2 --- 3 --- 4 --- 5 --- 6 --- 7 --- 8 --- 9 --- 10 --- 11 --- 12 --- 13
             │    │   histogram(2)=  0                 1                 10                  2
             │    │                <--- 'foo' COLLATE en_US_u_ks_level2 ---- 'world' COLLATE en_US_u_ks_level2
             │    ├── cost: 64.25
             │    ├── key: (1)
             │    ├── fd: (1)-->(2)
             │    ├── distribution: test
             │    └── prune: (1,2)
             └── filters
                  └── s:2 = 'hello' COLLATE en_US_u_ks_level2 [outer=(2), constraints=(/2: [/'hello' COLLATE en_US_u_ks_level2 - /'hello' COLLATE en_US_u_ks_level2]; tight), fd=()-->(2)]
            
[23:38:44] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/9508/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/collated_strings with config local: 3 tests, 1 failures
[23:38:45] --- total progress: 3 statements
--- total: 3 tests, 1 failures
    logic.go:4191: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/7c0b99e5fac8a43429506c705d588ce2/logTestExecBuild_collated_strings822237745
--- FAIL: TestExecBuild_collated_strings (2.25s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExecBuild_collated_strings.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/sql/opt/exec/execbuilder/tests/local/local_test.TestExecBuild_collated_strings [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/13605544?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/13605544?buildTab=artifacts#/) on master @ [eb42e39c887b9429cc46aed9db7ef66a4f6ebe82](https://github.com/cockroachdb/cockroach/commits/eb42e39c887b9429cc46aed9db7ef66a4f6ebe82):


```
             ├── key: (1)
             ├── fd: ()-->(2)
             ├── distribution: test
             ├── prune: (1)
             ├── scan t98400
             │    ├── columns: k:1 s:2
             │    ├── stats: [rows=13, distinct(1)=13, null(1)=0, distinct(2)=3, null(2)=0]
             │    │   histogram(1)=  0  1  0  1  0  1  0  1  0  1  0  1  0  1  0  1  0  1  0  1   0  1   0  1   0  1
             │    │                <--- 1 --- 2 --- 3 --- 4 --- 5 --- 6 --- 7 --- 8 --- 9 --- 10 --- 11 --- 12 --- 13
             │    │   histogram(2)=  0                 1                 10                  2
             │    │                <--- 'foo' COLLATE en_US_u_ks_level2 ---- 'world' COLLATE en_US_u_ks_level2
             │    ├── cost: 42.86
             │    ├── key: (1)
             │    ├── fd: (1)-->(2)
             │    ├── distribution: test
             │    └── prune: (1,2)
             └── filters
                  └── s:2 = 'hello' COLLATE en_US_u_ks_level2 [outer=(2), constraints=(/2: [/'hello' COLLATE en_US_u_ks_level2 - /'hello' COLLATE en_US_u_ks_level2]; tight), fd=()-->(2)]
        but found (query options: "") :
            select
             ├── columns: k:1 s:2
             ├── stats: [rows=10, distinct(2)=1, null(2)=0]
             │   histogram(2)=  0                 10
             │                <--- 'hello' COLLATE en_US_u_ks_level2
             ├── cost: 64.41
             ├── key: (1)
             ├── fd: ()-->(2)
             ├── distribution: test
             ├── prune: (1)
             ├── scan t98400
             │    ├── columns: k:1 s:2
             │    ├── stats: [rows=13, distinct(1)=13, null(1)=0, distinct(2)=3, null(2)=0]
             │    │   histogram(1)=  0  1  0  1  0  1  0  1  0  1  0  1  0  1  0  1  0  1  0  1   0  1   0  1   0  1
             │    │                <--- 1 --- 2 --- 3 --- 4 --- 5 --- 6 --- 7 --- 8 --- 9 --- 10 --- 11 --- 12 --- 13
             │    │   histogram(2)=  0                 1                 10                  2
             │    │                <--- 'foo' COLLATE en_US_u_ks_level2 ---- 'world' COLLATE en_US_u_ks_level2
             │    ├── cost: 64.25
             │    ├── key: (1)
             │    ├── fd: (1)-->(2)
             │    ├── distribution: test
             │    └── prune: (1,2)
             └── filters
                  └── s:2 = 'hello' COLLATE en_US_u_ks_level2 [outer=(2), constraints=(/2: [/'hello' COLLATE en_US_u_ks_level2 - /'hello' COLLATE en_US_u_ks_level2]; tight), fd=()-->(2)]
            
[02:05:07] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/13794/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/collated_strings with config local: 3 tests, 1 failures
[02:05:07] --- total progress: 3 statements
--- total: 3 tests, 1 failures
    logic.go:4191: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/7c0b99e5fac8a43429506c705d588ce2/logTestExecBuild_collated_strings3796282124
--- FAIL: TestExecBuild_collated_strings (2.29s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExecBuild_collated_strings.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/opt/exec/execbuilder/testdata/collated_strings
+++ b/pkg/sql/opt/exec/execbuilder/testdata/collated_strings
@@ -3,7 +3,7 @@
 # Regression test for incorrectly using the key-encoding for collated strings
 # as upper bounds of histogram buckets (#98400).
 statement ok
-CREATE TABLE t98400 (k INT PRIMARY KEY, s STRING COLLATE en_US_u_ks_level2);
+CREATE TABLE t98400 (k INT PRIMARY KEY, s STRING COLLATE en_US_u_ks_level2, FAMILY (k, s));
 INSERT INTO t98400 SELECT i, 'hello' FROM generate_series(1, 10) g(i);
 INSERT INTO t98400 SELECT i, 'world' FROM generate_series(11, 12) g(i);
 INSERT INTO t98400 SELECT 13, 'foo';

